torch/_export/utils.py

```diff
@@ -894,3 +905,25 @@ def _detect_fake_mode_from_gm(
                 fake_vals.append(fake_val)
 
     return detect_fake_mode(fake_inps + fake_vals)
+
+
+@contextmanager
+def _disabled_load_state_dict_hooks(mod: torch.nn.Module):
+    state_dict_hooks: Dict[int, Callable] = dict(mod._state_dict_hooks)
+    state_dict_pre_hooks: Dict[int, Callable] = dict(mod._state_dict_pre_hooks)
+    mod._state_dict_hooks.clear()
+    mod._state_dict_pre_hooks.clear()
+    try:
+        yield
+    finally:
+        mod._state_dict_hooks = state_dict_hooks
+        mod._state_dict_pre_hooks = state_dict_pre_hooks
+
+
+def _exported_program_state_dict(mod: torch.nn.Module):
+    """
+    Given a model, returns a state_dict that's consistent with exported program's
+    state_dict.
+    """
+    with _disabled_load_state_dict_hooks(mod):
+        return mod.state_dict()
```

Comment:
nit: I don't think we need this function, we can just tell ppl to directly use the disable hook?

Reply:
sure, removed now.